src/plugins/discourse/weights.js

```diff
@@ -1,9 +1,75 @@
 // @flow
 
 import {type NodeWeight} from "../../core/weights";
+import * as C from "../../util/combo";
+import * as NullUtil from "../../util/null";
 import {DEFAULT_TRUST_LEVEL_TO_WEIGHT} from "./createGraph";
 import {type User} from "./fetch";
 
+export type SerializedConfig = {|
+  +defaultWeight?: NodeWeight,
+  +weights?: {
+    [string]: NodeWeight,
+  },
+|};
+
+export type Config = {|
+  +defaultWeight: NodeWeight,
+  +weights: {
+    [string]: NodeWeight,
+  },
```

Comment:
I like the distinction between serialized and non-serialized configs. For the non-serialized configs, I recommend ES6 maps over object dicts. Pretty much the only reason we ever use dicts is because maps don't serialize.